public/doupload.php

```diff
@@ -25,7 +25,16 @@
 
 switch ($requestType) {
     case "uploadbadgeimage":
-        $response['Response'] = UploadBadgeImage($_FILES["file"]);
+        $uploadResponse = UploadBadgeImage($_FILES["file"]);
+        $response['Success'] = $uploadResponse['Success'];
+        unset($uploadResponse['Success']);
+
+        if ($uploadResponse['Error']) {
+            $response['Error'] = $uploadResponse['Error'];
+            unset($uploadResponse['Error']);
+        }
+
+        $response['Response'] = $uploadResponse;
```

Comment:
Is this kept for backwards compatibility in the clients or can we remove the `Response` key?
This would allow to shorten it to `$response = UploadBadgeImage($_FILES["file"])` to be in line with `dorequest` responses where `Response` is used as "data payload". In this case there is no actual data to be returned?

Reply:
Looking at `dorequest.php` I can see where this comes from with all those `Passthru`s. I wouldn't use that as a template as it's very inconsistent in `dorequest.php` and the more verbose variant of handling the response. I'd go simple by default in this case?

Reply:
Yes, this is very much for compatibility with the existing API usages.

https://github.com/RetroAchievements/RAIntegration/blob/552049a1b5d93b3bfed0c88529c3713330f4ca15/src/api/impl/ConnectedServer.cpp#L1388-L1401

The client could be modified to check for either, then we could eliminate it. `r=submitticket` has the same behavior.

Reply:
> In this case there is no actual data to be returned?

It's returning the ID associated to the new badge:
```
{"Success":true,"Response":{"Filename":"uploadbadgeimage.jpg","Size":4260,"BadgeIter":"61838"}}
```

Reply:
got it - sgtm then 👍